### ISSUE TITLE
Driver Signing.

### ISSUE THREAD

**Issue by basil00** [basil00/Divert#53]
I am looking for a new sponsor for driver signing.  The high-level requirements are:

* Willing to sign new releases of the `WinDivert32.sys` and `WinDivert64.sys` driver (probably about 1-2 releases per year).
* Have an Extended Validation (“EV”) Code Signing Certificate as required by [Windows 10](http://blogs.msdn.com/b/windows_hardware_certification/archive/2015/04/01/driver-signing-changes-in-windows-10.aspx).

Note that there is no immediate problem as the current release is already signed.  This is for anticipated future releases or bug fixes.

If you can help then please contact basil at reqrypt.org.

**Comment by TechnikEmpire** [comment]
It's looking like I'm going to be getting an EV cert to sign WinDivert for my own purposes within the next month or so. It's not 100% but I'll contact you when I'm at that point.

**Comment by basil00** [comment]
That'd be great.  Please let me know one way or the other.

**Comment by TechnikEmpire** [comment]
@basil any suggestions or advice on which cert to buy and where? I find lots of ev certs but some of them make different claims about code signing.

**Comment by basil00** [comment]
> I find lots of ev certs but some of them make different claims about code signing.

Usually you just want the cheapest that will do the job.  I found [this page](https://msdn.microsoft.com/en-us/library/windows/hardware/hh801887.aspx) that has a list of EV certificate authorities that should work (given they are directly linked to by Microsoft).

**Comment by TechnikEmpire** [comment]
@basil00 I remember some info you had about special rights for sponsors. Can you link to that info? I'm also curious if it's possible to add in some sort of mechanism to prevent tampering. For example, a content filter or AV scanner that uses WinDivert to capture all data, if there's something at the driver level where you can prevent a force shut down. This way you could open a capture all handle that drops all by default, and someone can't just run `sc.exe stop WinDivert1.XX` to get by it. Thoughts? 

Thanks for your time.

**Comment by basil00** [comment]
@TechnikEmpire, I am not sure what info you are referring to.  Perhaps contact me via email and we can discuss.

> For example, a content filter or AV scanner that uses WinDivert to capture all data, if there's something at the driver level where you can prevent a forced shut down of the driver. This way you could open a capture all handle that drops all by default, and someone can't just run sc.exe stop WinDivert1.XX to get by it. Thoughts?

This might not be a good idea as part of a general WinDivert release that can also be used by malicious applications, e.g. AdWare.  In such cases it is important that the user/Administrator be able to disable WinDivert as a last resort.

It might be okay for a specialized WinDivert that is locked down to a specific application (e.g. AV).  I am not sure how it can be implemented though.

**Comment by thalomatt** [comment]
@TechnikEmpire Do you know if you need an INF file to submit to the new driver signing standards?

**Comment by TechnikEmpire** [comment]
@thalomatt I'm not sure. I've never gone through the process myself.

**Comment by TechnikEmpire** [comment]
I'm in touch with an EV cert supplier. It's going to take some time because as a sole proprietor I need to jump through more hoops (signing oaths basically) but when I manage to get an EV I'll sign new releases.

**Comment by basil00** [comment]
Thanks, there really should be a new release that includes #92.  Also I have some other minor driver improvements on my todo list.  One thing I want to change is not having `WinDivertSend()` block on injection complete, which may be a performance bug.

**Comment by TechnikEmpire** [comment]
Yeah I know there's a few things emerging, plus I'd like to see if I can fork it and make some additions you may be interested in. That aside, I'm pretty heavily relying on your work here for my own open source project so I'd like to contribute back somehow if I can. It may be a bit but I am getting one and will definitely sign once I do.

**Comment by thalomatt** [comment]
@TechnikEmpire , 
When you've got your cert, let me know if I can help - it wasn't a trivial matter packaging it up for Microsoft to sign the driver.  At least not trivial to this non-Windows developer.  But seems to work for all of our users - a few Windows 7 users said it didn't work for them, but after asking if they had all the updates, they didn't reply, so I'm guessing they updated and it started working.

**Comment by basil00** [comment]
@thalomatt I'd guess the Windows 7 problem is because you built WinDivert with a newer version of WDK.  I assume this is a requirement from Microsoft?  The newer driver will work under Windows 7 only after the user has installed the latest updates.

**Comment by TechnikEmpire** [comment]
I believe the group I'm working with has an EV cert inbound. We had troubles with initial orders, the company ended up reneging on their offer to issue to an individual after the fact. Will update when it comes in.

**Comment by basil00** [comment]
@TechnikEmpire, that sounds promising.  I am also in contact with another company that has recently expressed interest in helping out.  The driver signing process is not as simple as before (now has hardware tokens, SHA1 versus SHA256, dev portal step), so I don't know how long it will take.

Also, it seems that Windows Server 2016 has even [stricter driver signing requirements](https://www.osr.com/blog/2016/06/02/driver-signing-details-emerge/), requiring HLK testing.

**Comment by basil00** [comment]
@thalomatt writes...

> But seems to work for all of our users - a few Windows 7 users said it didn't work for them, but after asking if they had all the updates, they didn't reply, so I'm guessing they updated and it started working.

Actually, I now think the problem is that you have signed the driver using the newer SHA2 algorithm, which is not supported by non-updated versions of Windows 7.  To fix this problem, you are supposed to "rekey" your EV code signing certificate to allow duel SHA1/SHA2 signing.  To rekey you need to contact your certificate provider.  Otherwise, you can not bother and just not support non-updated Windows 7.

**Comment by basil00** [comment]
There is a new release which contains EV certificate signed drivers: https://github.com/basil00/Divert/releases/tag/v1.3.0

It has also been signed by the Microsoft dev portal, so should work for Windows 10 with secureboot enabled.  I have not had the time to verify this since my virtualization environment doesn't support secureboot.

There are caveats:

* The signature uses the SHA2 algorithm which is not supported by Windows 7 unless a [patch](https://support.microsoft.com/en-us/help/3033929/microsoft-security-advisory-availability-of-sha-2-code-signing-support) is installed.  If your Windows 7 system is up-to-date the patch should already be installed.
* The drivers will not work for Windows Server 2016 unless you disable secure boot.  This is because even stricter driver signing requirements including passing the HLK tests.  This is on the TODO list.

Note also that version 1.3.0 is essentially the same as version 1.2.0-rc but with bug fixes.  The latest performance patches have not been included in this release.

**Comment by TechnikEmpire** [comment]
@basil00 Just curious, will we lose Vista compat with the new driver signing requirements?

**Comment by basil00** [comment]
Vista and unpatched Windows 7 support should be possible by "rekeying" the EV certificate to SHA1, then dual signing with SHA1/SHA2.  That said, Vista is no longer officially supported by Microsoft, so I do not intend to "officially" support it anymore either.

**Comment by nefarius** [comment]
Heya @basil00 I'd like to hop in on the driver signing train 😄

**Comment by basil00** [comment]
Great, if you want to help then contact me via email.

**Comment by joveice** [comment]
Status on this?

**Comment by basil00** [comment]
The project currently has two sponsors who have expressed willingness to sign the driver.  There has not been a new version for a while, but this may change in Q2 2018.

**Comment by arifulhuq** [comment]
@basil00 What is the status on this? I'd like to help as a sponsor and/or explore possibilities to fast track this work.

**Comment by basil00** [comment]
@arifulhuq If you'd like to help sign the driver then please contact me via email (see email [here](https://reqrypt.org/windivert.html)).  I'd certainly welcome more sponsors.

Currently the project has two sponsors but I've yet to contact them regarding WinDivert2.0, but probably will do soon.

**Comment by dhaavi** [comment]
I spent the last weeks/months working on a driver that is somewhat similar to WinDivert - it's geared more towards firewall applications and will be open sourced on Github soon. I am now trying to figure out how to sign this thing (got the EV cert!).

If anyone here would be willing to give a quick overview how to go about this, that would be great. Would also be nice to have this documented better for WinDivert. (We used the build environment of WinDivert.)
How have WinDivert signers done this in the past?

Anyway, as soon as I get the hang of this, I'd also be happy sign WinDivert releases.

(PM via [Twitter](https://twitter.com/dehaavi) or [Email](https://safing.io/contact/))

**Comment by basil00** [comment]
You'll need to go through all the Microsoft documentation on how to do it.  I've never done it personally.

The gist of it is that you need to create and sign (with your EV certificate) a driver package (sys + inf) and upload it do the dev portal for attestation signing.  Microsoft will then sign the driver and it is ready for use.

**Comment by dhaavi** [comment]
Yes, I have gone through the Microsoft documentation until my head hurt. 😏

I almost wanted to start with the attestation signing, but then I read [here](https://docs.microsoft.com/en-gb/windows-hardware/drivers/dashboard/attestation-signing-a-kernel-driver-for-public-release) that `An attestation signed driver will only work for Windows 10. It will not work for other versions of Windows, such as Windows 8.1, Windows 7, or any Windows Server versions.`

After checking with [Statcounter](http://gs.statcounter.com/os-version-market-share/windows/desktop/worldwide) and realizing that Win10 still only has a Windows market share of 55%, I though that it would be really great to at least also sign it for Win7.

Going further down that road, the docs also explain [here](https://docs.microsoft.com/en-gb/windows-hardware/drivers/dashboard/get-drivers-signed-by-microsoft-for-multiple-windows-versions) how to sign for all Windows versions:
`Use the Hardware Lab Kit (HLK) to test your submission against Windows 10 and use the Hardware Certification Kit (HCK) to test against earlier versions of Windows. [...] This is the only way to make a submission apply to all Windows versions.`

So the two questions that arose, were:
- Does attestation signing really only work for Win10? I mean, this is an internal network filter driver. There are compatibility guarantees across Windows versions, and there is no interaction with custom hardware.
- How much work is HLK/HCK testing? Is there any way to automate this? A DevOps pipeline maybe?

I should have been more clear in my first comment.

Also, the docs say that attestation signing `is more complicated, but still a valid option`. I have the feeling that HLK/HCK testing is way more complicated.

If this is going to much off topic for this issue, I'll happily move this somewhere else. I didn't think of this becoming a huge thread and hoped to ping the right people by commenting here.

**Comment by TechnikEmpire** [comment]
That's odd cause those docs are from 2017 yet I've gone through attestation signing and AFAIK this is how releases here are done. You zip it, upload it in the dashboard and Microsoft spits out a signed version. Its fast and easy. Just make sure you zip it right. 32 and 64 bit need to be in folders in the zip named the same way windivert names them. (Amd64 or whatever etc).

**Comment by dhaavi** [comment]
Thanks @TechnikEmpire, you are right. This actually worked, contrary to the official docs. I suspect that it works because Microsoft certifies the driver as `Universal`, because it only uses internal APIs.

The repo is now online, here is the description of the semi-automated signing process: https://github.com/Safing/portmaster-windows-kext/tree/master/release

To be explicit: I verified that the signature works on Windows 7, 8.1 and 10. I had to first install updates on the Win7 VM, as it was from 2015 or so.

I guess this means I am available for WinDivert signing now! 😉

**Comment by basil00** [comment]
@dhaavi 

> Does attestation signing really only work for Win10? I mean, this is an internal network filter driver. There are compatibility guarantees across Windows versions, and there is no interaction with custom hardware.

Strictly speaking, yes, but your driver will also be dual signed by your EV certificate, and the latter will work for Windows 7, 8.

> How much work is HLK/HCK testing? Is there any way to automate this? A DevOps pipeline maybe?

I've never tried it, but from what I read it seems like a hassle to set up.  Certainly much more work than attestation signing.

The only outstanding question is whether Windows Server 2016 accepts attestation signed drivers, or if a  WHQL release signature is required (i.e., HLK).  The docs state that HLK is required, but there is a lot of confusion on the topic, as it seems attestation signing also works fine.  From what I read it only makes a difference if the driver package uses a `.cat` file, which WinDivert does not.

> I guess this means I am available for WinDivert signing now!

Great, if you want to help sign, then please contact me via email (see bottom of page [here](https://reqrypt.org/windivert.html)).

**Comment by dhaavi** [comment]
> Strictly speaking, yes, but your driver will also be dual signed by your EV certificate, and the latter will work for Windows 7, 8.

My driver worked on Win7, Win8.1 and Win10 with just the Microsoft signature, but did not work with just my signature. I use the Microsoft signature as the primary one (Win7 only detects the primary signature), and then add mine afterwards for improved transparency.

> The only outstanding question is whether Windows Server 2016 accepts attestation signed drivers

I am currently downloading Windows Server 2016 and 2019, will report back.

**Comment by dhaavi** [comment]
The `Universal` driver signature also works on Windows Server 2016 and 2019. Both with only the Microsoft signature and with my EV sig added. 🎉 Nice!

**Comment by basil00** [comment]
Did you test with Secure Boot enabled?  Otherwise I think it will work without WHQL regardless.

**Comment by dhaavi** [comment]
Um. I don't know. I ran everything in Virtualbox and didn't turn anything off.
What would be the default for this environment?

> without WHQL

Do you mean that just my EV cert should work when Secure Boot is disabled?

**Comment by basil00** [comment]
@dhaavi I'm not sure if it is possible for VirtualBox to emulate secureboot for the guest.  This is part of the reason I never tried, since it might require a physical machine to test.

Btw, to what extent is PortmasterKext a derivative work of WinDivert?  Much of the code appears different but the README mentions "based on/influenced".  If there is some WinDivert code, it is important not to remove copyright notices at least for the relevant portions.

**Comment by nefarius** [comment]
That's one of the many caveats; a cross-signed driver will not load on Secure Boot enabled systems due to more restrictive Code Integrity Policies.

**Comment by dhaavi** [comment]
Thanks for the hint about Secure Boot. To clarify: the driver is signed directly by Microsoft and then also by me. The driver then has two distinct signatures.
I will investigate further and check if the driver is accepted if signed that way.

**Comment by nefarius** [comment]
That's another pitfall: since Secure Boot is available since Windows 8 the Windows 10 signature won't work there and your cross-signed cert will also be denied.

**Comment by ronshah90** [comment]
@dhaavi you can use Hyper-V to emulate Windows 2016 with Secure Boot.
According to this comment: https://github.com/MicrosoftDocs/windows-driver-docs/issues/1068#issue-382747909 it might actually work

**Comment by dhaavi** [comment]
Sorry, forgot to update you guys with the most recent findings:
I had my colleague test the signatures on his Secure Boot enabled Windows 10 machine - the signatures, both only from Microsoft and with mine added to it, worked fine. I don't have access to a machine with Win8.1 and Secure Boot at the moment.

So, to sum it up - I have two versions of the driver: one is only signed by Microsoft, the other has my EV signature added to it for transparency. Both work on:
- Windows 7
- Windows 8.1
- Windows 10 (also with Secure Boot)
- Windows Server 2016
- Windows Server 2019

Thanks for the hint about Hyper-V, I may take that route for future testing.

**Comment by basil00** [comment]
> Windows Server 2016

It'd be interesting to also try this one with Secure Boot.

**Comment by TechnikEmpire** [comment]
@basil00 It seems none of the drivers are dual signed with sha1 anymore. Is this intended? I assume that attempting to support non-updated Windows 7 is simply not a thing anymore.

**Comment by basil00** [comment]
Yes pretty much.  Also because the SHA1 signature seemed to cause more problems than it solved, such as mysterious revocation errors.

**Comment by helloray** [comment]
I have KB3033929 patch installed but run WinDivert-2.2.0-A/B/C with an error: failed to open the WinDivert device (577).  WinDivert-1.4.3-A is running well without any problem. Any suggestions? Thanks.

**Comment by basil00** [comment]
@helloray Try this version: https://reqrypt.org/download/WinDivert-2.2.0-D.zip

**Comment by helloray** [comment]
@basil00 Same result. error: failed to open WinDivert handle (err = 577)
![image](https://user-images.githubusercontent.com/5954203/78032753-130c0700-7398-11ea-92de-b83db41b07b6.png)

**Comment by basil00** [comment]
I forgot to mention that you should reboot before trying the new version.  If not, then Windows will attempt to reuse the previous driver with the signature it did not like.

**Comment by helloray** [comment]
Hi @basil00 , I have already rebooted before try WinDivert-2.2.0-D. Any other possible reasons?

**Comment by basil00** [comment]
I am not sure.  As a last resort, you can try manually deleting any `WinDivert` entry in the registry.  That seemed to have worked for other people.

**Comment by wumn290** [comment]
Stuck here. It's been a day. Who knows why
![RTX截图未命名1](https://user-images.githubusercontent.com/23630667/79041559-5d587800-7c23-11ea-9484-3be80f329965.png)

**Comment by muse117** [comment]
> I am not sure. As a last resort, you can try manually deleting any `WinDivert` entry in the registry. That seemed to have worked for other people.

The Version 2.2 A/B/C/D is double sha-256 signature. You should set a sha-1 and sha-256 signature. This can support Windows 7/ Windows 2008.

**Comment by basil00** [comment]
I usually ask the sponsors (who sign the driver) if they also want to support SHA1, but most do not bother.  It is still possible to run SHA256 drivers by using an up-to-date version of Windows 7, or at least by installing a patch: https://support.microsoft.com/en-us/help/3033929/microsoft-security-advisory-availability-of-sha-2-code-signing-support

**Comment by wumn290** [comment]
I signed the driver with our company's certificate, and then submitted it to Microsoft for signature. The choice is:
Windows 10 Client versions 1506 and 1511 (TH2)
Windows 10 Client versions 1506 and 1511 x64 (TH2)
Windows 10 Client version 1607 (RS1)
Windows 10 Client version 1607 x64 (RS1)
Windows 10 Client version 1703 Client (RS2)
Windows 10 Client version 1703 Client x64 (RS2)
Windows 10 Client version 1709 Client (RS3)
Windows 10 Client version 1709 Client x64 (RS3)
  I've been stuck in the Scanning stage, I don't know why

**Comment by TechnikEmpire** [comment]
IIRC just select a single target with the lowest version of windows 10. The portal scans the ini files and figures out itself what architectures you've included and such.

**Comment by wumn290** [comment]
I signed the driver with our company's certificate, and then submitted it to Microsoft for signature. The choice is:
Windows 10 Client versions 1506 and 1511 (TH2)
Windows 10 Client versions 1506 and 1511 x64 (TH2)
Windows 10 Client version 1607 (RS1)
Windows 10 Client version 1607 x64 (RS1)
Windows 10 Client version 1703 Client (RS2)
Windows 10 Client version 1703 Client x64 (RS2)
Windows 10 Client version 1709 Client (RS3)
Windows 10 Client version 1709 Client x64 (RS3)
  I've been stuck in the Scanning stage, I don't know why

**Comment by wumn290** [comment]
> IIRC just select a single target with the lowest version of windows 10. The portal scans the ini files and figures out itself what architectures you've included and such.

Does IIRC mean RS2? I now choose this way:
Windows 10 Client versions 1506 and 1511 (TH2)
Windows 10 Client versions 1506 and 1511 x64 (TH2)
Windows 10 Client version 1607 (RS1)
Windows 10 Client version 1607 x64 (RS1)
Give it a try

**Comment by wumn290** [comment]
Now the scan fails：
Scanning Notes
{"code":"4001","details":{"errorInfo":"ConfirmedMalware"},"innerError":null}

**Comment by TechnikEmpire** [comment]
No sorry, its just short for "if I recall correctly". The portal is finicky and will get stuck without explanation sometimes. Pick the simplest/base option. In this case, the lowest version of windows. Also make sure that you're only choosing attestation signing. Theres another type of signing where the portal runs a myriad of tests against the driver. You don't want that. Simply attestation signing.

**Comment by TechnikEmpire** [comment]
@basil00 that MS portal output needs your attention.
@wumn290 that doesn't look good. I've tagged the author, he will have to follow up.

**Comment by basil00** [comment]
> {"code":"4001","details":{"errorInfo":"ConfirmedMalware"},"innerError":null}

That is quite concerning.  If Microsoft decides to shadowbans the driver then that is pretty much the end of the project.

Some questions:

* What version of WinDivert did you attempt to sign?
* Did you make any modifications to the driver before signing?

Can anyone else with an EV certificate verify this for me?

If confirmed, the next step would be to contact Microsoft support to complain about a false positive.  WinDivert is not malware and should not be classified as such.

**Comment by wumn290** [comment]
> > {"code":"4001","details":{"errorInfo":"ConfirmedMalware"},"innerError":null}
> 
> That is quite concerning. If Microsoft decides to shadowbans the driver then that is pretty much the end of the project.
> 
> Some questions:
> 
> * What version of WinDivert did you attempt to sign?
> * Did you make any modifications to the driver before signing?
> 
> Can anyone else with an EV certificate verify this for me?
> 
> If confirmed, the next step would be to contact Microsoft support to complain about a false positive. WinDivert is not malware and should not be classified as such.

I am using WinDivert64.sys and WinDivert32.sys under WinDivert-1.4.3-A \ x86, which are not compiled from the source code, the driver is not changed before signing, but the inf is written by myself, other driver files of our company can The signature passed, but windivert failed

**Comment by basil00** [comment]
One of the sponsors resigned version 2.2.0 of the driver and had no problems.  So this might just be a false positive in Microsoft's malware detection that affects version 1.4.3 of the driver binary.

There are are few things you could try, such as upgrading to newer versions of the driver, recompiling the driver, or contacting Microsoft support to complain about the false positive.

**Comment by SizzlingCalamari** [comment]
> I have KB3033929 patch installed on Win7 but run WinDivert-2.2.0-A/B/C with an error: failed to open the WinDivert device (577). WinDivert-1.4.3-A is running well without any problem. Any suggestions? Thanks.

> @helloray Try this version: https://reqrypt.org/download/WinDivert-2.2.0-D.zip

@basil00 I had the same issue with A/B/C where they wouldn't load on a fully updated W7. The D version worked for me. Could you upload it to the project site or add it to releases? Thanks!

**Comment by helloray** [comment]
@SizzlingCalamari Your D version worked for me. Thanks

**Comment by basil00** [comment]
### Looking for a new sponsor for driver signing

WinDivert 2.2.1 is available but is currently unsigned.  If anyone can help with driver signing, please contact basil at reqrypt.org.

**Comment by Fplyth0ner-Combie** [comment]
> ### Looking for a new sponsor for driver signing
> WinDivert 2.2.1 is available but is currently unsigned. If anyone can help with driver signing, please contact basil at reqrypt.org.

We can help to sign driver. Contacted by email.

**Comment by basil00** [comment]
@Fplyth0ner-Combie Thanks very much for your help.

A WinDivert 2.2.1 release (with signed drivers) is now available here: https://github.com/basil00/Divert/releases/tag/v2.2.1

**Comment by sergeevabc** [comment]
Windows 7 x64 user here. I unpacked `x64` of [WinDivert 2.2.2-A](https://github.com/basil00/WinDivert/releases/tag/v2.2.2) to `C:\Test` and ran `test.exe`. It worked well, `passed = 100.00%`, but a few seconds later I got this…

![](https://github.com/user-attachments/assets/87101e63-43b9-49b2-a8fc-03305c5e1080)

Could you be so kind to fix it?

**Comment by sergeevabc** [comment]
Hello?

**Comment by teknixstuff** [comment]
> Windows 7 x64 user here. I unpacked `x64` of [WinDivert 2.2.2-A](https://github.com/basil00/WinDivert/releases/tag/v2.2.2) to `C:\Test` and ran `test.exe`. It worked well, `passed = 100.00%`, but a few seconds later I got this…
> 
> ![](https://private-user-images.githubusercontent.com/373718/425865969-87101e63-43b9-49b2-a8fc-03305c5e1080.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjkwNzkwMjUsIm5iZiI6MTc2OTA3ODcyNSwicGF0aCI6Ii8zNzM3MTgvNDI1ODY1OTY5LTg3MTAxZTYzLTQzYjktNDliMi1hOGZjLTAzMzA1YzVlMTA4MC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwMTIyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDEyMlQxMDQ1MjVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02ZjgyMzRmZWY0NWVhZTM1MzdiZDI0OGZiYWM2OWRhNTJiNzY2ZmU5NWVmMzI0OGVjNTc2ZDViODUxNzU0YjhiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.FxQRoqHi9hv_hv4-RkjZx2yIwHEZ46a-6b8Sv-nbyKk)
> 
> Could you be so kind to fix it?

This happens because the PCA service still doesn't understand SHA2, even with the SHA2 patch applied. The dialog is incorrect and can be ignored safely.
